src/coordinator/ts/m3db/block/converter_test.go

```diff
@@ -0,0 +1,170 @@
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package block
+
+import (
+	"testing"
+	"time"
+
+	"github.com/m3db/m3db/src/coordinator/models"
+	"github.com/m3db/m3db/src/dbnode/encoding"
+	"github.com/m3db/m3db/src/dbnode/ts"
+	"github.com/m3db/m3x/ident"
+	xtime "github.com/m3db/m3x/time"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func newSeriesBlock(seriesIterator encoding.SeriesIterator, start, end time.Time) SeriesBlock {
+	return SeriesBlock{
+		start:          start,
+		end:            end,
+		seriesIterator: seriesIterator,
+	}
+}
+
+func newMultiNamespaceSeries(ctrl *gomock.Controller, now time.Time, t *testing.T) ([]SeriesBlocks, []SeriesBlocks) {
+	rawTagsOne := []string{"foo", "bar", "same", "tag"}
+	seriesOneIterators := []encoding.SeriesIterator{
+		newMockIterator(t, "test_one", rawTagsOne, ctrl, []ts.Datapoint{}),
+		newMockIterator(t, "test_one", rawTagsOne, ctrl, []ts.Datapoint{}),
+	}
+	seriesOneBlockOne := newSeriesBlock(seriesOneIterators[0], now, now.Add(10*time.Minute))
+	seriesOneBlockTwo := newSeriesBlock(seriesOneIterators[1], now.Add(10*time.Minute), now.Add(20*time.Minute))
+	tagsOne, err := ident.NewTagStringsIterator(rawTagsOne...)
+	require.NoError(t, err)
+
+	seriesOne := SeriesBlocks{
+		Blocks: []SeriesBlock{seriesOneBlockOne, seriesOneBlockTwo},
+		ID:     ident.StringID("test_one"),
+		Tags:   tagsOne,
+	}
+
+	rawTagsTwo := []string{"biz", "baz", "same", "tag"}
+	seriesTwoIterators := []encoding.SeriesIterator{
+		newMockIterator(t, "test_two", rawTagsTwo, ctrl, []ts.Datapoint{}),
+		newMockIterator(t, "test_two", rawTagsTwo, ctrl, []ts.Datapoint{}),
+	}
+
+	seriesTwoBlockOne := newSeriesBlock(seriesTwoIterators[0], now, now.Add(10*time.Minute))
+	seriesTwoBlockTwo := newSeriesBlock(seriesTwoIterators[1], now.Add(10*time.Minute), now.Add(20*time.Minute))
+	tagsTwo, err := ident.NewTagStringsIterator(rawTagsTwo...)
+	require.NoError(t, err)
+
+	seriesTwo := SeriesBlocks{
+		Blocks: []SeriesBlock{seriesTwoBlockOne, seriesTwoBlockTwo},
+		ID:     ident.StringID("test_two"),
+		Tags:   tagsTwo,
+	}
+
+	return []SeriesBlocks{seriesOne}, []SeriesBlocks{seriesTwo}
+}
+
+func newMockIterator(t *testing.T, id string, tags []string, ctrl *gomock.Controller, dps []ts.Datapoint) encoding.SeriesIterator {
+	seriesIterator := encoding.NewMockSeriesIterator(ctrl)
+
+	seriesIterator.EXPECT().ID().Return(ident.StringID(id))
+
+	tagIter, err := ident.NewTagStringsIterator(tags...)
+	require.NoError(t, err)
+	seriesIterator.EXPECT().Tags().Return(tagIter)
+
+	seriesIterator.EXPECT().Next().Return(true).Times(len(dps))
+	seriesIterator.EXPECT().Next().Return(false)
+
+	for _, dp := range dps {
+		seriesIterator.EXPECT().Current().Return(ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value}, xtime.Millisecond, nil).Times(1)
```

Comment:
nit: might be a bit easier to read if you generate the point separately
`dp := ts.Datapoint{Timestamp: dp.Timestamp, Value: dp.Value}`